src/ppl/src/SimulatedAnnealing.cpp

```diff
@@ -177,19 +183,48 @@ void SimulatedAnnealing::randomAssignment()
                                                                 last_slot);
 
       int slot = distribution(generator_);
-      while (slots_[slot].used) {
+      int mirrored_slot;
+      bool free = !slots_[slot].used;
+      if (io_pin.isMirrored()) {
+        free = isFreeForMirrored(slot, mirrored_slot);
```

Comment:
Should this be free = free && isFreeForMirrored(slot, mirrored_slot); ?  Likewise below